OSDOCS#10502: Cluster extension status message improvements

michaelryanpeter opened the pull request:
<!--- PR title format: [GH#<gh-issue-id>][BZ#<bz-issue-id>][OCPBUGS#<jira-issue-id>][OSDOCS#<jira-issue-id>]: <short-description-of-the-pr> --->

<!--- If your changes apply to the latest release and/or in-development version of OpenShift, open your PR against the `main` branch.

* For more details about the information requested in this template, see:
  https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/create_or_edit_content.adoc#submit-PR --->

Version(s): 4.16
<!--- Specify the version or versions of OpenShift your PR applies to. -->

Issue: [OSDOCS-10502](https://issues.redhat.com//browse/OSDOCS-10502) and https://issues.redhat.com/browse/OSDOCS-9021
<!--- Add a link to the Bugzilla, Jira, or GitHub issue, if applicable. --->

Link to docs preview: 
* https://77274--ocpdocs-pr.netlify.app/openshift-enterprise/latest/operators/olm_v1/olmv1-installing-an-operator-from-a-catalog.html#olmv1-installing-an-operator_olmv1-installing-operator
* https://77274--ocpdocs-pr.netlify.app/openshift-enterprise/latest/operators/olm_v1/olmv1-installing-an-operator-from-a-catalog.html#olmv1-updating-an-operator_olmv1-installing-operator

<!--- Add direct link(s) to the exact page(s) with updated content from the preview build. --->

QE review:
- [x] QE has approved this change.
<!--- QE approval is required to merge a PR except for changes that do not impact the meaning of the docs. --->

Additional information:
<!--- Optional: Include additional context or expand the description here.--->

<!--- After you open your PR, ask for review from the OpenShift docs team:
  For community authors: Tag @openshift/team-documentation in a GitHub comment.--->

adellape commented on modules/olmv1-installing-an-operator.adoc:
```diff
@@ -87,14 +88,29 @@ select( .schema == "olm.channel" ) | select( .name == "latest" ) | \
+
 ----
 ====
+. If you want to install your extension into a new namespace, run the following command.
```
```suggestion
. If you want to install your extension into a new namespace, run the following command:
```

michaelryanpeter commented:
Thank you, @jianzhangbjz. Comments addressed in https://github.com/openshift/openshift-docs/pull/77274/commits/7852f3f54dba3399da5969fb77eec4548ea7b485

varshaprasad96 commented on modules/olmv1-installing-an-operator.adoc:
```diff
@@ -142,48 +160,90 @@ clusterextension.olm.operatorframework.io/pipelines-operator created
+`spec.channel`:: Displays the channel defined in the CR of the extension.
+`spec.version`:: Displays the version or version range defined in the CR of the extension.
+`status.conditions`:: Displays information about the status and health of the extension.
+`type: Deprecated`:: Displays if the extension is deprecated. The value of `False` in the `status` field indicates that the extension is not deprecated.
```
```suggestion
`type: Deprecated`:: Displays if the extension is deprecated. The value of `False` in the `status` field indicates that the bundle which is to be installed is not deprecated.
```

I'm not sure if "extension" is deprecated would be right. It's more on the lines of - while extension tries to install a bundle, the bundle is itself deprecated. I'll leave the wording to you, as you feel it fit :)

joelanford replied:
For `type: Deprecated`, could we somehow say it's a rollup of `PackageDeprecated`, `ChannelDeprecated`, and `BundleDeprecated`, and let the other descriptions do the heavy lifting?

michaelryanpeter replied:
@varshaprasad96 would something like this make sense?
```suggestion
`type: Deprecated`:: Displays if the extension is deprecated. The value of `False` in the `status` field indicates that the resolved bundle is not deprecated.
```

joelanford commented on modules/olmv1-installing-an-operator.adoc:
```diff
@@ -142,48 +160,90 @@ clusterextension.olm.operatorframework.io/pipelines-operator created
+`type: Deprecated`:: Displays if the extension is deprecated. The value of `False` in the `status` field indicates that the extension is not deprecated.
+`type: PackageDeprecated`:: Displays if the package is deprecated. The value of `False` in the `status` field indicates that the extension is not deprecated.
+`type: ChannelDeprecated`:: Displays if the channel is deprecated. The value of `False` in the `status` field indicates that the extension is not deprecated.
+`type: BundleDeprecated`:: Displays if the channel is deprecated. The value of `False` in the `status` field indicates that the extension is not deprecated.
```
Copy pasta

```suggestion
`type: BundleDeprecated`:: Displays if the bundle is deprecated. The value of `False` in the `status` field indicates that the extension is not deprecated.
```

michaelryanpeter commented:
@joelanford and @varshaprasad96 Thanks for the feedback! I addressed your comments in https://github.com/openshift/openshift-docs/pull/77274/commits/d68653e19ebbf0eb07ade4d39acff7f91d0b04c4

I think that this will be clearer in 4.17 where I plan on making a `status.conditions` table. Let me know what you think.

michaelryanpeter commented on modules/olmv1-installing-an-operator.adoc:
```diff
@@ -142,48 +160,90 @@ clusterextension.olm.operatorframework.io/pipelines-operator created
+`spec.channel`:: Displays the channel defined in the CR of the extension.
+`spec.version`:: Displays the version or version range defined in the CR of the extension.
+`status.conditions`:: Displays information about the status and health of the extension.
+`type: Deprecated`:: Displays if the extension is deprecated. The value of `False` in the `status` field indicates that none ofthe `reason: Deprecated` conditions are marked as deprecated.
```
```suggestion
`type: Deprecated`:: Displays if the extension is deprecated. The value of `False` in the `status` field indicates that none of the other `reason: Deprecated` conditions are marked as deprecated.
```

michaelryanpeter commented:
Thank you for the feedback, @jianzhangbjz , @joelanford and @varshaprasad96 

I have addressed your collective comment` in this commit: https://github.com/openshift/openshift-docs/pull/77274/commits/95125c985f1a2353e8dd69529116d4bc4a42b1e3

kuiwang02 commented on modules/olmv1-installing-an-operator.adoc:
```diff
@@ -142,48 +160,95 @@ clusterextension.olm.operatorframework.io/pipelines-operator created
+    uid: c54237be-cde4-46d4-9b31-d0ec6acc19bf
+  spec:
+    channel: latest
+    installNamespace: openshift-marketplace
```
@michaelryanpeter suggest openshift-operators, not openshift-marketplace.
per my understanding, openshift-marketplace is used for catsrc, while openshift-operators is used for operator installation.

should check it in the PR.

michaelryanpeter replied:
@kuiwang02 Thank you for the feedback. Addressed in https://github.com/openshift/openshift-docs/commit/587f6bd25e3cad42aefa47f211480b2064cb1fb4.

kuiwang02 commented on modules/olmv1-installing-an-operator.adoc:
```diff
@@ -142,48 +160,95 @@ clusterextension.olm.operatorframework.io/pipelines-operator created
+  metadata:
+    annotations:
+      kubectl.kubernetes.io/last-applied-configuration: |
+        {"apiVersion":"olm.operatorframework.io/v1alpha1","kind":"ClusterExtension","metadata":{"annotations":{},"name":"pipelines-operator"},"spec":{"channel":"latest","installNamespace":"openshift-marketplace","packageName":"openshift-pipelines-operator-rh","pollInterval":"30m"}}
```
@michaelryanpeter it is not changed here `"installNamespace":"openshift-marketplace"`.

michaelryanpeter commented:
/label peer-review-needed

varshaprasad96 reviewed (approved):
The changes look good to me!

/lgtm

michaelryanpeter commented:
/label peer-review-needed

joelanford commented:
/lgtm

kuiwang02 commented:
i am ok

jianzhangbjz commented:
/lgtm

michaelryanpeter commented:
/cherrypick enterprise-4.16
